web3/_utils/transactions.py

```diff
@@ -65,11 +66,22 @@
     "chainId",
 ]
 
+if TYPE_CHECKING:
+    from web3 import Web3  # noqa: F401
+
+
+def _get_gas_price(w3: "Web3", transaction: TxParams) -> Optional[Wei]:
+    gas_price = w3.eth.generate_gas_price(transaction)
+    if gas_price is None:
+        return w3.eth.gas_price
+    return gas_price
+
+
 TRANSACTION_DEFAULTS = {
     "value": 0,
     "data": b"",
     "gas": lambda w3, tx: w3.eth.estimate_gas(tx),
-    "gasPrice": lambda w3, tx: w3.eth.generate_gas_price(tx) or w3.eth.gas_price,
```

Comment:
Nice catch 👍 I think it would make sense before EIP1559. Right now given the logic in `fill_transaction_defaults` it does not. So I will remove that and convert it to simple lambda expression without `w3.eth.gas_price`